index.js

```diff
@@ -129,6 +120,11 @@ arg.flag = fn => {
 	return fn;
 };
 
+arg.of = fn => (value, name, prev = []) => {
+	prev.push(fn(value, name, prev[prev.length - 1]));
+	return prev;
+};
+
```

Comment:
This function doesn't make any sense to me. This seems so specific that I think users should be writing their own logic for whatever this is for - not included as a standard handler.

Plus it's not in the scope of this PR.

Reply:
This function needs to exist regardless - it’s how this works internally by wrapping the `[Handler]` behaviour. Do you just want me to avoid exporting it?

Reply:
Why is it necessary? Can you explain what it does? Why keep track of each iterative value? I'm lost as to what this logic does.

Reply:
It turned your previous logic of `[Handler]` into a function wrapper so it's not implemented within the loop. It just creates an array and calls the function each time appending to the array. This works so that `[String]` continues to function as before, but now the logic is all encapsulated within `of(String)` - it doesn't need to copied twice to build features like https://github.com/zeit/arg/pull/33.

Reply:
It works the same as this chunk before:

```
if (isArray) {
  if (result[argName]) {
    result[argName].push(value);
  } else {
    result[argName] = [value];
  }
}
```

Upon calling the function, it creates an array. Then it calls `Handler` each time (e.g. `Boolean`, `String`, `Number`, `myHandler`) with the value and appends it to the array. It proxies through the `name` and `prev` values as though it were a regular function call in `type(...)`.

Reply:
Ah, yes please keep this internal. In fact, since it's used once, just inline it.

Further, the confusion stemmed from `prev[prev.length - 1]` - that just needs to be `prev`.

Reply:
> it doesn't need to copied twice to build features like #33.

With #33, I don't see how that would help anything, honestly.

Reply:
In #33 I needed to use the type function when writing into `—` outside the arg loop. That’s where the changes come from originally. 

I can remove the `prev` change since it’s internal, but it is an odd signature because it’s compatible with TypeScript - you receive `T[]` in prev but expect to return `T`. I was looking at keeping the signature consistent with non-array handling types. What are your thoughts given the type information?

Reply:
Also, what do you mean by inline it? Like put it inside of the loop instead of outside the function?

Reply:
> I'm sorry, I think I still don't understand

You're only using `arg.of` [here](https://github.com/zeit/arg/pull/34/files#diff-168726dbe96b3ce427e7fedce31bb0bcR34). That's the only place it's being used. Just replace the call to `arg.of` with the contents of `arg.of` directly, hence _inlining_ it, since there's no reason to expose it in the API.

Reply:
Inlined 👍